src/examples/euler/EulerSingleSidedHook.sol

```diff
@@ -0,0 +1,126 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.24;
+
+import {IERC20} from "forge-std/interfaces/IERC20.sol";
+import {IPoolManager} from "v4-core/src/interfaces/IPoolManager.sol";
+
+import {FixedPointMathLib} from "solmate/src/utils/FixedPointMathLib.sol";
+import {PoolKey} from "v4-core/src/types/PoolKey.sol";
+import {Currency} from "v4-core/src/types/Currency.sol";
+import {TickMath} from "v4-core/src/libraries/TickMath.sol";
+
+import {JIT} from "../../JIT.sol";
+import {JITHook} from "../../JITHook.sol";
+
+import {EulerUtils} from "./EulerUtils.sol";
+
+/// @title Euler Account JIT Hook
+/// @notice Pulls funds from a specified Euler vaults through a specific swap account, create a JIT position, and transfers funds back to the swap account
+/// @notice When user is doing exaactIn or exactOut swap, this will JIT single-sided liquidity only
+/// @dev JIT position is 8% width
+contract EulerSingleSidedHook is JITHook, EulerUtils {
+    constructor(address evcAddr, IPoolManager poolManagerAddr, address vaultA, address vaultB, address swapAccountAddr)
+        JITHook(poolManagerAddr)
+        EulerUtils(evcAddr, vaultA, vaultB, swapAccountAddr)
+    {}
+
+    /// @inheritdoc JITHook
+    function _pull(PoolKey calldata key, IPoolManager.SwapParams calldata swapParams)
+        internal
+        override
+        returns (address excessRecipient, uint128 amount0, uint128 amount1)
+    {
+        excessRecipient = swapAccount;
+
+        (, Currency outputCurrency, uint256 amount) = _getInputOutputAndAmount(key, swapParams);
+
+        // transferFrom: depositor's currency0 or currency1 to the PoolManager based on currency-out from the pool
+        if (outputCurrency == key.currency0) {
+            amount0 = _sendToPoolManager(vault0, key.currency0, uint128(amount));
+        } else {
+            amount1 = _sendToPoolManager(vault1, key.currency1, uint128(amount));
+        }
+    }
+
+    /// @inheritdoc JITHook
+    function _recipient() internal view override returns (address) {
+        return swapAccount;
+    }
```

Comment:
my understanding of euler is a bit limited, but does sending capital to a swap account mark it as collateral?

just thinking about v4's sync/send/settle design, and was wondering if there was an similar pattern in Euler; i.e. an additional function needs to be poked to signal that transferred-capital should be marked as _something_